src/extensions/agent_modules/cloud_communication/communication_channel/mqtt_broker/src/adu_communication_channel.c

```diff
@@ -151,55 +151,107 @@ static inline void Log_Debug_With_Frequency(int freqSeconds, const char* format,
 ADU_MQTT_COMMUNICATION_MGR_STATE* CommunicationManagerStateFromModuleHandle(ADUC_AGENT_MODULE_HANDLE commHandle)
 {
     ADUC_AGENT_MODULE_INTERFACE* interface = (ADUC_AGENT_MODULE_INTERFACE*)commHandle;
-    if (!interface)
+    if (interface == NULL)
+    {
         return NULL;
-    ADU_MQTT_COMMUNICATION_MGR_STATE* state = (ADU_MQTT_COMMUNICATION_MGR_STATE*)interface->moduleData;
-    return state;
+    }
+
+    return (ADU_MQTT_COMMUNICATION_MGR_STATE*)interface->moduleData;
 }
 
 /**
- * @brief Create the communication channel module.
+ * @brief Create and Init Communication Mgr State object
+ *
+ * @return ADU_MQTT_COMMUNICATION_MGR_STATE* The new communication manager state instance, or NULL on failure.
+ * @remark Caller is responsible for calling DestroyCommunicationMgrState on the returned state instance.
  */
-ADUC_AGENT_MODULE_HANDLE ADUC_Communication_Channel_Create()
+static ADU_MQTT_COMMUNICATION_MGR_STATE* CreateCommunicationMgrState()
 {
-    ADUC_AGENT_MODULE_INTERFACE* interface = NULL;
     ADU_MQTT_COMMUNICATION_MGR_STATE* state = NULL;
 
-    ADUC_ALLOC(state);
+    ADU_MQTT_COMMUNICATION_MGR_STATE* tmp = calloc(1, sizeof(*tmp));
+    if (tmp == NULL)
+    {
+        goto done;
+    }
 
-    state->pendingSubscriptions = VECTOR_create(sizeof(ADUC_MQTT_SUBSCRIBE_CALLBACK_INFO));
-    if (state->pendingSubscriptions == NULL)
+    tmp->pendingSubscriptions = VECTOR_create(sizeof(ADUC_MQTT_SUBSCRIBE_CALLBACK_INFO));
+    if (tmp->pendingSubscriptions == NULL)
     {
-        Log_Error("Failed to create pending subscriptions vector");
         goto done;
     }
 
-    Log_Debug("pendingSubscriptions obj created @ %p", state->pendingSubscriptions);
+    tmp->connectionRetryParams = defaultConnectionRetryParams;
+
+    state = tmp;
+    tmp = NULL;
+
+done:
+    free(tmp);
```

Comment:
similar here, would be nice to see a Alloc_ and Free_MqttCommunicationMgrState()